### PR TITLE
merge dev to master

### PR DESCRIPTION
**Pull request by laveesingh**
**In progress**

Following are the changes to be mentioned:

- fixed slideWidth calculation approximation
- fixed unusual scrolls in focusOnSelect mode
- added appendDots method for customization of dots
- modified logic for handling odd/even cases where there were unusual scrolls in opposite direction
- implemented unslick feature properly
- fixed variableWidth issues like blank spaces at edges, improper alignment
- handling focus loss in case of fade=true
- responsive lazyloading bug fixed
- increased verticalswiping resistance from 4 to 10

### CODE REVIEW

File: src/dots.js [hunk 1]
```diff
@@ -59,10 +59,6 @@ export class Dots extends React.Component {
       );
     });
 
-    return (
-      <ul className={this.props.dotsClass} style={{display: 'block'}}>
-        {dots}
-      </ul>
-    );
+    return React.cloneElement(this.props.appendDots(dots), {className: this.props.dotsClass});
```
Comment (akiran) [inline src/dots.js]: @laveesingh Do we need cloneElement here ?

We can do some thing like this
```
const AppendDots = this.props.appendDots
return <AppendDots className={this.props.dotsClass} />
```

Comment (laveesingh) [reply src/dots.js]: I'll try and replace that.

File: src/mixins/event-handlers.js [hunk 2]
```diff
@@ -301,21 +316,22 @@ var EventHandlers = {
...
+          this.setState({ currentDirection: 0 }) // unverified fix of above line
           break;
 
         case 'right':
         case 'up':
           newSlide = this.state.currentSlide - this.getSlideCount();
           slideCount = this.props.swipeToSlide ? this.checkNavigable(newSlide) : newSlide;
-          this.state.currentDirection = 1;
+          // this.state.currentDirection = 1; // critical: change this line with setState statement
+          this.setState({ currentDirection: 1 }) // unverified fix of above line
```
[the remark this answers is not in the document]
Comment (laveesingh) [reply src/mixins/event-handlers.js]: absolutely

**Comment by dijonkitchen** [comment]
FYI, this likely does the following:

Fixes #94
Fixes #221
Fixes #556

**Pull request merged**